src/main/java/de/tum/in/www1/artemis/repository/hestia/CoverageReportRepository.java

```diff
@@ -33,7 +33,7 @@ default CoverageReport findByIdElseThrow(Long coverageReportId) {
             AND (s.type <> 'ILLEGAL' OR s.type IS NULL)
             order by s.submissionDate desc
             """)
-    List<CoverageReport> getLatestCoverageReportsForProgrammingExercise(@Param("programmingExerciseId") Long programmingExerciseId, Pageable pageable);
+    List<CoverageReport> getLatestCoverageReportsForLegalSubmissionsForProgrammingExercise(@Param("programmingExerciseId") Long programmingExerciseId, Pageable pageable);
```

Comment:
I'm not quite sure if this is correct. Your selection above is not illegal or not set. That does not equal legal or does it?

Reply:
I am also not really familiar with this topic of Artemis. But in the `ProgrammingsubmissionRepository` this term is used for this exact same query (`findLatestLegalSubmissionForParticipation`).
So I assume there is some reason for this.

Reply:
Sorry, forgot to answer. I think we don't have to focus on it now.